popup.js

```diff
@@ -1,30 +1,44 @@
-var months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
-var days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"]
+/**
+ * Over-engineered top-level closure to avoid
+ * anything here from leaking globally.
+ * @param {HTMLElement} timeEl
+ * @param {HTMLElement} dateEl
+ */
+;((timeEl, dateEl) => {
+  ;('use strict')
 
-function clockWrite() {
-	//Set all the time variables
-	var d = new Date(),
-	h = d.getUTCHours(),
-	m = d.getUTCMinutes(),
-	s = d.getUTCSeconds();
-	
-	//Set all the date variables
-	day	= d.getUTCDay(),
-	date = d.getUTCDate(),
-	month = d.getUTCMonth(),
-	year = d.getUTCFullYear();
-	
-	h = (h < 10 ? '0' : '') + h;
-	m = (m < 10 ? '0' : '') + m;
-	s = (s < 10 ? '0' : '') + s;
-	
-	utctime = (h + ':' + m + ':' + s);
-	utcdate = (days[day] + ' ' + months[month] + ' ' + date + ', ' + year);
-	
-	document.getElementById("time").firstChild.nodeValue = utctime;
-	document.getElementById("date").firstChild.nodeValue = utcdate;
-}
+  /** @type {Intl.DateTimeFormatOptions} */
+  const dateFormatOpts = {
+    weekday: 'short',
+    day: 'numeric',
+    month: 'short',
+    year: 'numeric',
+    timeZone: 'UTC',
+  }
 
-document.addEventListener('DOMContentLoaded', function () {
-	setInterval(clockWrite, 1000); clockWrite();
-});
+  /** @type {Intl.DateTimeFormatOptions} */
+  const timeFormatOpts = {
+    hour12: false,
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+    timeZone: 'UTC',
+  }
```

Comment:
`Date` formatting back in the day was an ugly, tedious thing. It's still not amazing, but thanks to the [`Intl`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat) standard making it into modern browsers everywhere, there is some relief. This builtin Intl object essentially creates a bunch of ultra-specific options for how a number or date should be displayed.  These option objects are then passed to functions that format dates (see lines 33 or 37 below).  So, for example, if you set the `hour` option to `'2-digit'`, the formatted string returned will add a leading `0` if needed to conform with the `'2-digit'` requirement.